### ISSUE TITLE
Bus stops

### ISSUE THREAD

**Issue by bgo-eiu** [ZeLonewolf/openstreetmap-americana#436]
Blocked by #387 

As busways (#308) and guided busways (#413) have been brought up, it seems appropriate to open a separate bus stop issue as well. (In line with this, bus lanes would probably end up being a different topic too, since these each would require different implementations.)

Based on some conversations on Slack, I think it would be helpful to clarify the role of bus features within a street map. Many of the first reference points that come to mind for print maps of bus transit in America are schematic transit maps, which typically show some degree of simplification of a route or network, and may take on an abstract form or show an overlay on a geographically accurate street grid. However, these types of maps often benefit from de-emphasizing other features in some way, and can be considered specialized. It's not the case that bus passengers are a niche audience, but that these maps only serve a particular use case even for bus passengers. Most people who rely on public transport also use some form of generalized street map, because it contains the details they need to make connections to and from specific stops, context for wayfinding, and the shape of the street network. These features get omitted from schematic transit maps, because that including that information density is incompatible with the goal of highlighting the general shape and connections in a network at a glance. To that end, fulfilling the role of a good street map useful for public transit users seems like a fitting role for Americana.

The better examples of transit maps do make some important and interesting design decisions that I still think would be helpful to draw influence from even if it doesn't make sense to replicate their role directly. They are also some of the easier to find reference points for print maps of bus-related features.  Maps which [distinguish the frequent network from the coverage network](https://humantransit.org/basics/the-case-for-frequency-mapping) can make a big difference in how usable they are for example. Unfortunately bus transit is historically neglected despite quietly constituting a very substantial portion of local travel in American cities, so there are not as many examples of good street maps for bus passengers, or even good general guide maps featuring bus stops. It is common to find maps which make some really unhelpful or questionable decisions in how they show bus stops and/or routes.

For symbology, I quite like the idea of a silhouette of the MUTCD standard bus icon. Many transit agencies simply use the icon at stops in place of a custom logo, and many custom logos are just minimal variants to it, so it has a broad recognizability. It is also common for signposts for park and rides along roadways to feature this icon, so it seems like it would integrate well with the existing route sign aesthetic. A previous older PR took a similar approach at #17 that got me thinking about this.
![image](https://user-images.githubusercontent.com/100172442/174459442-5795ed39-3e24-4436-98bc-85545e5ef40c.png)

The simplest first step would just be to render bus stops where they exist, before getting into more details on what else we can do with them. At the lowest zoom levels, all of them should be visible. The zoom level bus stops appear at should be z12 or lower if I understand the order correctly - when service roads appear, because they very often cluster around service roads at bus bays or drop off zones or even parking lots.

Once they are on the map in some capacity, it would be neat if we were able to collapse bus stops at intermediate zoom levels to make the locations of the significant ones more clear. Showing one stop for each "stop area group" relation, or omitting stops which aren't part of a frequent route as tagged by `interval` could be some ways to do that.

**Comment by ZeLonewolf** [comment]
I think we need to come to a _general_ agreement about whether we prefer shielded POI icons like this, or ones with no background, as appear in osm-carto.  This decision will be the foundation for #387 and open the door to general POI contribution.

**Comment by claysmalley** [comment]
Our shield guidelines have backgrounds simplified to a 1px stroke. I don't see why POI icons shouldn't be the same.

**Comment by 1ec5** [comment]
> Once they are on the map in some capacity, it would be neat if we were able to collapse bus stops at intermediate zoom levels to make the locations of the significant ones more clear.

The [Transit](https://wiki.openstreetmap.org/wiki/Transport_Map) tile layer attempts to group stop areas together. I’m not sure how much processing is involved, but it would probably require some postprocessing for the vector tiles, because GL JS doesn’t support clustering features from vector tiles as it does for GeoJSON sources.

**Comment by bgo-eiu** [comment]
To be clear the standard sign I posted is just one variant, there are many signs which don't feature an outline, or which have a circle border, or other small modifications. The green square seems to be common for road-facing park and ride signs in particular. I'll find some other examples to share here for reference.

I'm relatively agnostic as to whether or not the symbol has an outline, and it's possible somewhere down the line we could have some variations in the symbology based on different types of bus related features. Carto actually uses both a symbol without an outline, and one with an outline, with the outline being reserved for bus stations specifically:

![Screenshot_20220618-124042](https://user-images.githubusercontent.com/100172442/174489444-7e3363ef-69ce-44ea-8ca6-60a5fe6076f9.png)

You can see it in the top right there at this bus station I've been working on mapping. I think it looks a bit weird to have it shown as one big bus symbol with lots of little bus stop children, so I was thinking stations would be where we might use agency logos. 

I don't feel too strongly about it, but I am slightly leaning towards the idea of a 1px outline for bus stops under the logic that they're often representing the position of a sign, and it seems intuitive for them to have a sign-like symbol. I'm not sure if a decision has been made about whether or not it's OK to have a mix of outlined and outline-less symbols, but in my experience there are some icons it makes sense for and others it doesn't. The box with a flag symbol commonly used for schools on paper maps is usually without an outline, but restroom symbols almost always have an outline. (I think that makes sense, especially since a restroom is a "contained" feature there's some logic in making it look contained.)

Are we sticking to 16x16 for POI symbols, or is there a size threshold? That will have some effect on choosing a symbol that looks legible.

**Comment by bgo-eiu** [comment]
Re: stop group areas, I've seen the Transit layer does it, but unfortunately it appears to be closed source. I was planning on looking at the code which reads road route relations for shields as a reference point, as I was thinking a process for displaying a symbol for a stop group could look similar.

I think it would be an interesting form of mapper feedback as well, since they're ostensibly useful but hard to see the results of. As far as I can tell, the Transit layer just shows bubbles around them, which is a form of feedback, but it makes the map more crowded when one of the main advantages of grouping stop features together is to simplify them without losing information about unrelated stops in conjunction.

**Comment by wmisener** [comment]
Rejuvenating this discussion now that some icons are beginning to be rendered (though we still lack many that are probably more important than bus stops). One possibility I've seen that I like is some kind of symbology of a stop, like a sign, or a sign next to a bus. See for inspiration, perhaps, the second and third icons suggested for maki here: https://github.com/mapbox/maki/issues/401
![image](https://user-images.githubusercontent.com/58491489/218820462-4e6babcb-aecc-4922-a257-58faa378ce69.png) ![image](https://user-images.githubusercontent.com/58491489/218820498-8604fde9-2c17-4da0-8794-236984fae81d.png)

Another enticing possibility could be to render attributes of bus stops, like shelters and benches. These attributes can be useful for route planning and wayfinding, and it would provide nice mapper feedback. In general, rendering attributes as variations on a main theme is a nice way to streamline icons, as nicely outlined by Andy Townsend [in this carto comment](https://github.com/gravitystorm/openstreetmap-carto/issues/3635#issuecomment-1146454522). One project that does this in a nice way is [Röntgen](https://wiki.openstreetmap.org/wiki/R%C3%B6ntgen). See for example how that project combines shelter, bench, and bus stop iconology: https://github.com/enzet/map-machine/blob/main/doc/bus_stop.png.
![image](https://user-images.githubusercontent.com/58491489/218821291-2982f90c-ed58-4607-b4ef-0af626a59e6a.png)

I think having a shielded/MUTCD-style sign for bus stations, similar to how Carto distinguishes them, would be a good approach.
